### DISCUSSION TITLE
Amateur's questions

### DISCUSSION THREAD

**Discussion by defazioa** [kachurovskiy/nanoels#144] [Ideas]
Hello, I'm unfamiliar with the forum so please excuse any missteps in protocol. I read your discussion with Jesse in his ESPels repository and, after trying to make heads or tails of the code, I very much liked your ideas with regard to organization and features. I looked for a long time before settling on Jon Bryan's "Atomic ELS" for the Mega 2560. I am almost completely unskilled in this area, I am more toolmaker than developer, but I have managed to get the thing working here in the garage on my little 250mm lathe using a closed loop NEMA 24 stepper, an encoder at 720ppr (2*360 due to gearing), and some Tylenol. Problem is;I don't like the AVR boards. I have used the ESP32 for other projects and would like it's speed for interrupt timing overhead and wifi for OTA updating but the ESP els project seemed incomplete and over-wrought at the same time, but I'm not experienced enough to modify it without help. I don't particularly like the wifi control, I'm using a Nextion display as you suggested, and I gutted the feed gearbox which now houses one 3 position maintained switch and one 3 position momentary, now operated by the two gearchange levers. It would be perfect if the maintained could be feed direction and the momentary was rapid, the only problem is that I don't know what I'm doing. Then to complicate matters I come here and see you're working on power feed. I have a milling machine too, so now the projects list is growing. I think what you fellows were talking about was more feature rich than I need, what I really want is for the lathe to behave as it would have with a quick change box and a working feed direction lever. If you did any work towards breaking that project up into more intuitive blocks as you suggested, I would take another shot at making it do what I need. My apologies for the diatribe, I'm sure you are busy, these are only wants. The Mega thing works, although I wish it used the existing spindle hall sensor for indexing and syncing, threadforms require a lot of extra attention to line up on successive passes. That and a little backlash compensation would be nice. Give me some time before you start working on surface grinder traverse and feed, I've got one of those too.      Thanks, **Aaron**

**Comment by kachurovskiy** [comment]
Hi Aaron, sorry, perhaps this is the wrong forum. This is NanoEls H2 and H4 project.

> I read your discussion with Jesse in his ESPels repository

Apologies but I don't recall such discussions, perhaps you could share a link?

> I have managed to get the thing working here in the garage on my little 250mm lathe using a closed loop NEMA 24 stepper, an encoder at 720ppr (2*360 due to gearing), and some Tylenol

Congratulations, please share the pictures of your machine if you get a chance for context and inspiration.

> I'm using a Nextion display as you suggested

I don't recall suggesting Nextion to anyone - perhaps you have a link to the discussion - or maybe you're thinking about another ELS project, not NanoEls?

**Reply by defazioa** [reply]
My apologies, not only the wrong forum, but the wrong contributor as well. I believe the individual I was looking for is named Chris. Thank you, I will post some photos when I find the right place. I'm very partial to the ESP32 and if I can find a solution I will gladly do it all over again. That machine has been apart and back together many times, it's an import so that's almost a requirement. Thank you again, I considered buying your ELS as well, best of luck to you with that, it looks very complete.

**Reply by digiexchris** [reply]
that's likely me. Here's the link to the issues on that repo: https://github.com/jschoch/ESPels/issues

**Comment by defazioa** [comment]
Hi Chris, I'm sure it's you. Thanks for reaching out, I had some ideas to
make the project make more sense from a user point of view. I am
unsatisfied with what I have now even though I just installed it, the guy's
name is Jon Bryan and the repository is on Github. If you find time I'd
like to discuss it with you. If you use my work email you'll get faster
responses, I rarely check this thing.

Thank you for taking the time to send a note, I didn't expect it really.

A.
